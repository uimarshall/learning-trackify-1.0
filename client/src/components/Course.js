import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export default function Subject({ subject: { image, name, id }, handleClick, mySubjects }) {
  return (
    <div className="subject-list-item bg-white">
      <div>
        <FontAwesomeIcon icon={image.split(' ')} />
      </div>
      <div className="text-sm">
        <div><span>{name}</span></div>
      </div>
      {
        mySubjects.includes(id) ? '' : (
          <div style={{ marginLeft: 'auto' }}>
            <button
              type="button"
              className="btn"
              onClick={() => handleClick(id)}
            >
              Register
            </button>
          </div>
        )
      }
    </div>
  );
}

Subject.propTypes = {
  mySubjects: PropTypes.arrayOf(PropTypes.number).isRequired,
  subject: PropTypes.shape({
    image: PropTypes.string,
    name: PropTypes.string,
    id: PropTypes.number,
  }).isRequired,
  handleClick: PropTypes.func.isRequired,
};
