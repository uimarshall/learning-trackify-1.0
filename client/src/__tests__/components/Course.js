import '@testing-library/jest-dom';

import * as React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer'
import Course from '../../components/Course';

jest.mock('../../components/Course', () => () => <div id="Course">Course</div>);
describe('Course Component', () => {
  test('renders Course in the DOM', () => {
    const { container } = render(<Course />);
    const mockComponent = container.querySelector('div#Course');

    expect(mockComponent).toBeInTheDocument();
    expect(mockComponent).not.toHaveTextContent(/^arbitrary$/); // to match the whole content
    expect(mockComponent).not.toHaveTextContent(/fire$/i); // to use case-insensitive match
    expect(mockComponent).toHaveTextContent('Course');
  });
});

describe('Course Component', () => {
  afterEach(cleanup);

  it('should take a snapshot for Course Component', () => {
    const asFragment = renderer.create(<Course />).toJSON()
    expect(asFragment).toMatchSnapshot();
  });
});