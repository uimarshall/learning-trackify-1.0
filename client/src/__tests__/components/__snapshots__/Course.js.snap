// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course Component should take a snapshot for Course Component 1`] = `
<div
  id="Course"
>
  Course
</div>
`;
